### PR TITLE
sample_ppc bug fix

### PR DESCRIPTION
**Pull request by junpenglao**
My fix in #2725 to use all chains breaks the `sample_ppc([point]...)` and also the progress bar. These issue should be fix here now and also add test for sample_ppc from a list.

### CODE REVIEW

File: pymc3/sampling.py [hunk 1]
```diff
@@ -971,7 +972,11 @@ def sample_ppc(trace, samples=None, model=None, vars=None, size=None,
         posterior predictive samples.
     """
     len_trace = len(trace)
-    nchain = trace.nchains
+    try:
+        nchain = trace.nchains
+
```
Comment (aloctavodia) [inline pymc3/sampling.py]: unnecessary space

File: pymc3/sampling.py [hunk 2]
```diff
@@ -1073,12 +1083,20 @@ def sample_ppc_w(traces, samples=None, models=None, weights=None,
     trace = []
     for i, j in enumerate(n):
         tr = traces[i]
-        len_trace = len(tr) 
-        nchain = tr.nchains 
+        len_trace = len(tr)
+        try:
+            nchain = tr.nchains
```
Comment (aloctavodia) [inline pymc3/sampling.py:1088]: Why the try/except? `tr.nchains` works even for traces with only 1 chain, right?

**Review by aloctavodia — APPROVED** [review]

File: pymc3/sampling.py (hunk 2 above)
Comment (junpenglao) [reply pymc3/sampling.py:1088]: Not work if you pass a list, it doesnt have `.nchains`. For example, you can do:
```python
with model:
    map1 = pm.find_MAP()
ppc = pm.sample_ppc([map1], sample=100, model=model)
```

Comment (aloctavodia) [reply pymc3/sampling.py:1088]: but `sample_ppc_w` is always expecting a list of traces

Comment (junpenglao) [reply pymc3/sampling.py:1088]: yeah I was thinking the same, but it is a bit of a corner case as `sample_ppc` allows a list, and sample_ppc_w should as well, something like:
`sample_ppc_w([[map1], [map2]], [model1, model2])`
what do you think?

Comment (aloctavodia) [reply pymc3/sampling.py:1088]: You are right, a general solution is a better option.
Could you add a line in the doctring of `sample_ppc_w`, something similar as what you did for `sample_pcc`?

**Pull request merged**